storey/table.py

```diff
@@ -109,29 +129,32 @@ async def _aggregate(self, key, data, timestamp):
             raise self._flush_exception
         if not self._schema:
             await self._get_or_save_schema()
-        cache_item = self._get_aggregations_attrs(key)
-        cache_item.aggregate(data, timestamp)
-        self._changed_keys.add(get_hashed_key(key))
+        async with self._get_lock(key):
+            cache_item = self._get_aggregations_attrs(key)
+            cache_item.aggregate(data, timestamp)
+            self._changed_keys.add(get_hashed_key(key))
 
     async def _get_features(self, key, timestamp):
         if self._flush_exception is not None:
             raise self._flush_exception
         if not self._schema:
-            await self._get_or_save_schema()
+            async with self._get_schema_lock():
+                await self._get_or_save_schema()
```

Comment:
better move the lock inside the function
_aggregate calls _get_or_save_schema as well